src/components/RestuarantContainer.js

```diff
@@ -38,7 +38,14 @@ const RestuarantContainer = (props) => {
                     <Shimmer/>
                 </>
                 :
-                filteredListOfRestuarants.map(restuarantData => <RestuarantCard key={restuarantData.info.id} restuarantData={restuarantData}/>)
+                filteredListOfRestuarants.map(restuarantData => 
+                    <Link
+                        to={`/restuarant/${restuarantData.info.id}`}
+                        key={restuarantData.info.id}
+                    >
+                        <RestuarantCard restuarantData={restuarantData}/>
+                    </Link>
+                )
```

Comment:
Implementation of Dynamic routing using Link component available in react-router-dom